### PR TITLE
Added orientation tolerance to MoveAction and 2 new unit tests

### PR DESCRIPTION
**Pull request by EvanMorcom**
### Description

Added optional orientation threshold to MoveAction.
The current implementation will set MoveAction as done based solely on the proximity to the destination and ignores the final orientation parameter 

### Testing Done
Added unit tests for new functionality, all existing tests pass.
### Resolved Issues
NA
### Review Checklist
- [X] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [X] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [X] **Remove all commented out code**
- [X] **Remove extra print statements**: for example, those just used for testing
- [X] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [X] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/thunderbots/software/ai/hl/stp/tactic/penalty_kick_tactic.cpp [hunk 1]
```diff
@@ -165,27 +164,30 @@ void PenaltyKickTactic::calculateNextIntent(IntentCoroutine::push_type& yield)
...
             yield(approach_ball_move_act.updateStateAndGetNextIntent(
-                *robot, behind_ball, (-behind_ball_vector).orientation(), 0));
+                *robot, behind_ball, (-behind_ball_vector).orientation(), 0, true));
         }
         else
         {
             const Point next_shot_position = evaluate_next_position();
             const Angle next_angle = (next_shot_position - ball.position()).orientation();
-
+            printf("\nTryingToRotate");
```
Comment (henryymliu) [inline src/thunderbots/software/ai/hl/stp/tactic/penalty_kick_tactic.cpp]: remove?

File: src/thunderbots/software/ai/hl/stp/tactic/penalty_kick_tactic.cpp [hunk 2]
```diff
@@ -165,27 +164,30 @@ void PenaltyKickTactic::calculateNextIntent(IntentCoroutine::push_type& yield)
...
+        else if (!approach_ball_move_act.done())
         {
             // The default behaviour is to move behind the ball and face the net
+            printf("\nDistanceFromBall=%f", (ball.position() - robot->position()).len());
+            printf("\nWhatWereSupposedToBe=%f", (behind_ball - robot->position()).len());
+            printf("\nDegreesOff=%f", robot.value()
+                    .orientation()
+                    .minDiff((-behind_ball_vector).orientation())
+                    .toDegrees() );
+            printf("\nTryingToMove");
```
Comment (henryymliu) [inline src/thunderbots/software/ai/hl/stp/tactic/penalty_kick_tactic.cpp]: remove?

**Review by henryymliu — CHANGES_REQUESTED** [review]
couple stray print statements, otherwise LGTM

**Review by henryymliu — APPROVED** [review]

**Comment by MathewMacDougall** [comment]
@EvanMorcom 	Can this be udpated and merged?

**Comment by MathewMacDougall** [comment]
@EvanMorcom I think this is good to go once it's updated from `upstream/master`

File: src/thunderbots/software/ai/hl/stp/tactic/penalty_kick_tactic.cpp (hunk 1 above)
Comment (EvanMorcom) [reply src/thunderbots/software/ai/hl/stp/tactic/penalty_kick_tactic.cpp]: Removed

File: src/thunderbots/software/ai/hl/stp/tactic/penalty_kick_tactic.cpp (hunk 2 above)
Comment (EvanMorcom) [reply src/thunderbots/software/ai/hl/stp/tactic/penalty_kick_tactic.cpp]: Removed

**Comment by EvanMorcom** [comment]
@MathewMacDougall Good to merge if you want to look it over quickly again.

**Comment by MathewMacDougall** [comment]
@EvanMorcom 	Ship it :ship:

**Review by MathewMacDougall — APPROVED** [review]

**Pull request merged**